### ISSUE TITLE
[Bug]: cn crashed by invalid memory address or nil pointer dereference and mo hung duaring runing sysbench delete test  on eks with 2*cn

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#9363]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):nightly-3e22243
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

duaring runing sysbench delete test  on eks with 2*cn for regression test, one of cn crash by panic:
{"level":"DEBUG","time":"2023/05/10 04:02:05.132762 +0000","caller":"disttae/reader.go:175","msg":"read [id __mo_rowid] with [0 65535]"}
{"level":"DEBUG","time":"2023/05/10 04:02:05.132842 +0000","caller":"disttae/reader.go:193","msg":"`block merge reader` operator catch a batch, batch length is 4761\n[vec-0(id)[type is INT, 0xc9b415f000] : len is 4761]\n[vec-1(__mo_rowid)[type is ROWID, 0xc9b415f080] : len is 4761]\n"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1e69416]

goroutine 53638983 [running]:
github.com/matrixorigin/matrixone/pkg/container/batch.(*Batch).SetAttributes(...)
        /go/src/github.com/matrixorigin/matrixone/pkg/container/batch/batch.go:200
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*blockMergeReader).Read(0xc48cc808f0, {0x2dd96d8?, 0xc9948f4b70?}, {0xc8969d9d20, 0x2, 0x2}, 0xc24d793aa0?, 0x47cc05?, {0x2db9a78, 0xc24bd6c000})
        /go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/disttae/reader.go:176 +0x3f6
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*mergeReader).Read(0xc82ce9e768, {0x2dd96d8, 0xc9948f4b70}, {0xc8969d9d20, 0x2, 0x2}, 0xc24d793b08?, 0xc24d793b38?, {0x2db9a78, 0xc24bd6c000})
        /go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/disttae/reader.go:213 +0x10c
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*mergeReader).Read(0xc82ce9e750, {0x2dd96d8, 0xc9948f4b70}, {0xc8969d9d20, 0x2, 0x2}, 0x7f52451341d8?, 0x30?, {0x2db9a78, 0xc24bd6c000})
        /go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/disttae/reader.go:213 +0x10c
github.com/matrixorigin/matrixone/pkg/vm/pipeline.(*Pipeline).Run(0xc24ed09c90, {0x2db9780, 0xc82ce9e750}, 0xc24bd6c000)
        /go/src/github.com/matrixorigin/matrixone/pkg/vm/pipeline/pipeline.go:79 +0x2a7
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).Run(0xc48cc80340, 0x2dd96d8?)
        /go/src/github.com/matrixorigin/matrixone/pkg/sql/compile/scope.go:75 +0x14e
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).ParallelRun(0xc48cc80340, 0xca61c4fc00, 0x0)
        /go/src/github.com/matrixorigin/matrixone/pkg/sql/compile/scope.go:267 +0xa78
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).RemoteRun(0xc48cc80340, 0xca61c4fc00)
        /go/src/github.com/matrixorigin/matrixone/pkg/sql/compile/scope.go:154 +0xb8
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun.func3(0xc24a805c20?)
        /go/src/github.com/matrixorigin/matrixone/pkg/sql/compile/scope.go:97 +0x26
created by github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun
        /go/src/github.com/matrixorigin/matrixone/pkg/sql/compile/scope.go:97 +0x54d

and then the mo hung:

<img width="1072" alt="image" src="https://github.com/matrixorigin/matrixone/assets/47846308/ccb48f42-581e-453d-ab11-9aa480429a16">

the cn panic log:
[cn-panic.log](https://github.com/matrixorigin/matrixone/files/11438264/cn-panic.log)

the profile of cn0 and cn 1:
[profile.tar.gz](https://github.com/matrixorigin/matrixone/files/11438272/profile.tar.gz)

### Expected Behavior

MO should run stablly.

### Steps to Reproduce

```Markdown
1、run a mo cluster on eks
2、run the following test cases Sequentially
   oltp point select 100,500,1000 threads
   oltp update 00,500,1000 threads
   oltp insert 00,500,1000 threads
   oltp delete 00,500,1000 threads
can see the github aciton
https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4932093333/jobs/8815191325
```

**Comment by xzxiong** [comment]
@LeftHandCold pls a first look

**Comment by xzxiong** [comment]
@aptend ptal
![image](https://github.com/matrixorigin/matrixone/assets/3927687/d827282d-37a7-4ab9-850e-f3def47e4027)

thx @LeftHandCold helps

**Comment by aptend** [comment]
> fixed in #9369

merged. please check @aressu1985

**Comment by aressu1985** [comment]
testing

**Comment by aressu1985** [comment]
fixed:fad1eaeace72fe6bacac673181020420a21d48ca
